File: project03/LetterGrades.java
/* LetterGrades.java is a driver for function LetterGrades
 *
 * Begun by: Prof. Adams, for CS 214 at Calvin College.
 * Haim Hong
 * Jan 27 2022
 ****************************************************************/

import java.util.Scanner;

public class LetterGrades {

   public static void main(String[] args) {
      for (int i = 100; i > 50; i-=3){ // looping until i is below 50
         System.out.println("\nTesting grade: " + i); // prints out the Grade its testing currently
         System.out.println(letterGrade(i)); // display its proper gRADE
      }
   }
   public static char letterGrade(int average)
   {
      switch (average / 10)
      {
         case 10: case 9: // checkes for each cases to find what matches average and returns right Letter grade for it
            return 'A';
         case 8:
            return 'B';
         case 7:
            return 'C';
         case 6:
            return 'D';
         default:
            return 'F';
      }
   }
}

File: project03/clojure/src/letter_grades.clj
;;;; driver for function letter_Grades

;;;; gets grades and decide letter grades for it
;;;; output the letter grade from the input

;;;; Haim Hong
;;;; Jan 27 2022
;;;;;;;;;;;;;;;;
(ns letter_grades)

(defn letterGrades [grade]
  (cond
    (= (quot grade 10) 10) "A" ;;;; after diving and it equals to the number then it outputs the right letter grade for it
    (= (quot grade 10) 9) "A"
    (= (quot grade 10) 8) "B"
    (= (quot grade 10) 7) "C"
    (= (quot grade 10) 6) "D"
    :else       "F"))  ;;;; when nothing matches then F

(defn -main []
    (loop [x 100] ;; looping until it goes below 50
          (when  (> x 50)
            (print "\nTesting grade: " x) (flush) ;; prints out the grade its testing for
            (println "\n" (letterGrades x)) ;; prints out the result
            (recur(- x 3))))     ; recursion to go back to loop
)

